showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITApiVersionHeaders.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.showcase.v1beta1.it;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.gax.httpjson.*;
+import com.google.common.collect.ImmutableList;
+import com.google.showcase.v1beta1.*;
+import com.google.showcase.v1beta1.it.util.TestClientInitializer;
+import io.grpc.*;
+import java.util.ArrayList;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ITApiVersionHeaders {
+  private static final String SPLIT_TOKEN = "&";
+  private static final String API_VERSION_HEADER_STRING = "x-goog-api-version";
+  private static final String HTTP_RESPONSE_HEADER_STRING =
+      "x-showcase-request-" + API_VERSION_HEADER_STRING;
+  private static final Metadata.Key<String> API_VERSION_HEADER_KEY =
+      Metadata.Key.of(API_VERSION_HEADER_STRING, Metadata.ASCII_STRING_MARSHALLER);
+
+  // Implement a client interceptor to retrieve the trailing metadata from response.
+  private static class GrpcCapturingClientInterceptor implements ClientInterceptor {
+    private Metadata metadata;
+
+    @Override
+    public <RequestT, ResponseT> ClientCall<RequestT, ResponseT> interceptCall(
+        MethodDescriptor<RequestT, ResponseT> method, final CallOptions callOptions, Channel next) {
+      ClientCall<RequestT, ResponseT> call = next.newCall(method, callOptions);
+      return new ForwardingClientCall.SimpleForwardingClientCall<RequestT, ResponseT>(call) {
+        @Override
+        public void start(Listener<ResponseT> responseListener, Metadata headers) {
+          Listener<ResponseT> wrappedListener =
+              new SimpleForwardingClientCallListener<ResponseT>(responseListener) {
+                @Override
+                public void onClose(Status status, Metadata trailers) {
+                  if (status.isOk()) {
+                    metadata = trailers;
+                  }
+                  super.onClose(status, trailers);
+                }
+              };
+
+          super.start(wrappedListener, headers);
+        }
+      };
+    }
```

Comment:
Not something that has to be done in this PR, but since using client interceptors is something that multiple tests have implemented, I think we can try and migrate this to TestClientInitializer if possible (I don't know if there are an limitations with that, but just something to consider).

Reply:
mmh, sounds a good idea. These interceptors are slightly different, but I guess we can create an interceptor that does it all.
To keep current pr from being cluttered, I would file a separate issue to look into and refactor these.